Library/src/Multiplayer/SpaceEntity.cpp

```diff
@@ -1131,32 +1131,45 @@ void SpaceEntity::DestroyComponent(uint16_t Key)
 
 ComponentBase* SpaceEntity::FindFirstComponentOfType(ComponentType Type, bool SearchDirtyComponents) const
 {
-	auto& CheckComponents = *GetComponents();
+	auto& CheckComponents							  = *GetComponents();
+	const csp::common::Array<uint16_t>* ComponentKeys = Components.Keys();
 
-	for (int i = 0; i < Components.Size(); ++i)
+	int i							= 0;
```

Comment:
Why are you declaring 'i' here and not in the for loop itself?

Reply:
Old habit, refactored as suggested